rust/azure_iot_operations_connector/src/filemount/observability_artifacts.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Types for extracting Observability configurations from an AIO deployment.
+//! Note that this module is a stopgap implementation and will be replaced in
+//! the future by a unified approach to observability endpoints.
+
+use std::path::PathBuf;
+
+/// Values extracted from the observability artifacts in an AIO deployment.
+pub struct ObservabilityArtifacts {
+    /// OTEL grpc/grpcs metric endpoint.
+    pub grpc_metric_endpoint: Option<String>,
+    /// OTEL grpc/grpcs log endpoint.
+    pub grpc_log_endpoint: Option<String>,
+    /// OTEL grpc/grpcs trace endpoint.
+    pub grpc_trace_endpoint: Option<String>,
+    /// Path to the directory containing trust bundle for 1P grpc metric collector.
+    pub grpc_metric_collector_1p_ca_mount: Option<PathBuf>,
+    /// Path to the directory containing trust bundle for 1P grpc log collector.
+    pub grpc_log_collector_1p_ca_mount: Option<PathBuf>,
+
+    /// OTEL http/https metric endpoint.
+    pub http_metric_endpoint: Option<String>,
+    /// OTEL http/https log endpoint.
+    pub http_log_endpoint: Option<String>,
+    /// OTEL http/https trace endpoint.
+    pub http_trace_endpoint: Option<String>,
+
+    /// OTEL 3P metric endpoint.
+    pub metric_endpoint_3p: Option<String>,
+    /// OTEL 3P metric export interval.
+    pub metric_export_interval_3p: Option<i32>,
```

Comment:
I think most of our other values like this are u64s or u32s, any reason for i32 here?